apps/mobile/src/components/ProfileView/ProfileViewSharedInfo/ProfileViewSharedCommunities.tsx

```diff
@@ -0,0 +1,175 @@
+        __typename
+        sharedCommunities(first: $sharedCommunitiesFirst, after: $sharedCommunitiesAfter)
+          @connection(key: "UserSharedInfoFragment_sharedCommunities") {
+          edges {
+            node {
+              __typename
+              ... on Community {
+                __typename
+                name
+                chain
+                contractAddress {
+                  address
+                  chain
+                }
+              }
+            }
+          }
+          pageInfo {
+            total
+          }
+        }
+        ...ProfileViewSharedCommunitiesSheetFragment
+      }
+    `,
+    userRef
+  );
+
+  const sharedCommunities = useMemo(() => {
+    const list = user.sharedCommunities?.edges?.map((edge) => edge?.node) ?? [];
+    return removeNullValues(list);
+  }, [user.sharedCommunities?.edges]);
+
+  const totalSharedCommunities = user.sharedCommunities?.pageInfo?.total ?? 0;
+
+  // Determine how many users to display by username
+  const communitiesToDisplay = useMemo(() => {
+    // In most cases we display a max of 2 communities. ie "community1, community2 and 3 others"
+    // But if there are exactly 3 shared communities, we display all 3. ie "community1, community2 and community3"
+    const maxNamesToDisplay = totalSharedCommunities === 3 ? 3 : 2;
+    return sharedCommunities.slice(0, maxNamesToDisplay);
+  }, [sharedCommunities, totalSharedCommunities]);
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+  const handleSeeAllPress = useCallback(() => {
+    bottomSheetRef.current?.present();
+  }, []);
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const handleCommunityPress = useCallback(
+    (contractAddress: ContractAddress) => {
+      const { address, chain } = contractAddress ?? {};
+
+      if (!address || !chain) return;
+      navigation.push('Community', {
+        contractAddress: address,
+        chain,
+      });
+    },
+    [navigation]
+  );
+
+  const content = useMemo(() => {
+    // Display up to 3 communities
+    const result = communitiesToDisplay.map((community) => {
+      if (!!community.contractAddress && community.chain) {
+        return (
+          <InteractiveLink
+            onPress={() =>
+              community.contractAddress && handleCommunityPress(community.contractAddress)
+            }
+            key={community.name}
+          >
+            <Typography
+              className="text-xs underline text-shadow"
+              font={{ family: 'ABCDiatype', weight: 'Regular' }}
+            >
+              {community.name}
+            </Typography>
+          </InteractiveLink>
+        );
+      }
+      return (
+        <Typography
+          className="text-xs underline text-shadow"
+          font={{ family: 'ABCDiatype', weight: 'Regular' }}
+        >
+          {community.name}
+        </Typography>
+      );
+    });
```

Comment:
nit: Get `key` prop warning. Might be related to the `.map` here

<img width="468" alt="CleanShot 2023-06-16 at 10 37 41@2x" src="https://github.com/gallery-so/gallery/assets/4480258/750f0152-4af4-4966-9049-a51bced909ad">

Reply:
fixed ✅